src/nodes/chunk_append/transform.c

```diff
@@ -39,65 +39,66 @@
 Expr *
 ts_transform_cross_datatype_comparison(Expr *clause)
 {
-	clause = copyObject(clause);
-	if (IsA(clause, OpExpr) && list_length(castNode(OpExpr, clause)->args) == 2)
+	if (!IsA(clause, OpExpr) || list_length(castNode(OpExpr, clause)->args) != 2)
 	{
-		OpExpr *op = castNode(OpExpr, clause);
-		Oid left_type = exprType(linitial(op->args));
-		Oid right_type = exprType(lsecond(op->args));
+		return clause;
+	}
 
-		if (op->opresulttype != BOOLOID || op->opretset == true)
-			return clause;
+	OpExpr *op = castNode(OpExpr, clause);
+	Oid left_type = exprType(linitial(op->args));
+	Oid right_type = exprType(lsecond(op->args));
 
-		if (!IsA(linitial(op->args), Var) && !IsA(lsecond(op->args), Var))
-			return clause;
+	if (op->opresulttype != BOOLOID || op->opretset == true)
+		return clause;
 
-		if (DATATYPE_PAIR(left_type, right_type, TIMESTAMPOID, TIMESTAMPTZOID) ||
-			DATATYPE_PAIR(left_type, right_type, TIMESTAMPTZOID, DATEOID))
-		{
-			char *opname = get_opname(op->opno);
-			Oid source_type, target_type, opno, cast_oid;
+	if (!IsA(linitial(op->args), Var) && !IsA(lsecond(op->args), Var))
+		return clause;
 
-			/*
-			 * if Var is on left side we put cast on right side otherwise
-			 * it will be left
-			 */
-			if (IsA(linitial(op->args), Var))
-			{
-				source_type = right_type;
-				target_type = left_type;
-			}
-			else
-			{
-				source_type = left_type;
-				target_type = right_type;
-			}
+	if (DATATYPE_PAIR(left_type, right_type, TIMESTAMPOID, TIMESTAMPTZOID) ||
+		DATATYPE_PAIR(left_type, right_type, TIMESTAMPTZOID, DATEOID))
+	{
+		char *opname = get_opname(op->opno);
+		Oid source_type, target_type, opno, cast_oid;
 
-			opno = ts_get_operator(opname, PG_CATALOG_NAMESPACE, target_type, target_type);
-			cast_oid = ts_get_cast_func(source_type, target_type);
+		/*
```

Comment:
The code looks close to what we have in `ts_transform_cross_datatype_comparison`. Have you considered splitting the function into smaller ones that can be re-used?

Reply:
This is the `ts_transform_cross_datatype_comparison`, do you mean it's similar to some other function? I'm not actually changing it much, just the formatting have changed because I transformed the first condition into early exit.